src/codesy_home.coffee

```diff
@@ -4,9 +4,25 @@ if $(".installed").length > 0
   $(".install-step").hide()
   $(".installed").show()
 
-console.time "codesy: check token"
+console.time("codesy: check token");
+
 if $("#api_token_pass").length > 0
-  console.log "codesy: found token"
-  chrome.storage.local.set auth_token: $("#api_token_pass").val()
-console.timeEnd "codesy: check token"
+  token_found = $("#api_token_pass").val()
+  if token_found isnt "" 
+    console.log("codesy: found token");
+    new_token = 
+      'domain': window.location.href,
+      'token': token_found
+      
+    chrome.storage.local.get (data) ->
+      data.domains = data.domains || []
+      isSaved = data.domains
+        .map((item) -> item.domain)
+          .indexOf(new_token.domain);
+      if isSaved isnt 0
+        if isSaved isnt -1
```

Comment:
Let's take out the `isnt 0` check - it's safe to replace the token in place and we'll need to do that if we change the token on the backend. I.e., "We changed your token, just go back to codesy.io and your widget will get the new token."